apps/consulting/posts/openai-pseudocode.md

```diff
@@ -0,0 +1,276 @@
+╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+What code feedback would you like to provide? (leave empty for approval): 
+```
+
+`pseudocode` generates appropriate prompts to the LLM asking it to
+provide code in response to the user's requirements. In this case we
+specified `review=True` in the decorator which prompted `pseudocode`
+to ask for user feedback on the generated code.
+
+```shell
+What code feedback would you like to provide? (leave empty for approval): this code is messy can you simplify it?
+╭─────────────────────────────────────────────────────────── LLM Feedback ─────────────────────────╮
+│ this code is messy can you simplify it?                                                          │
+╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
+╭──────────────────────────────────────────────────────────── Code Review ─────────────────────────╮
+│                                                                                                  │
+│ import os                                                                                        │
+│ import datetime                                                                                  │
+│ import requests                                                                                  │
+│                                                                                                  │
+│ def run(repository: str) -> list:                                                                │
+│     headers = {'Authorization': f"Bearer {os.environ['GITHUB_TOKEN']}"}                          │
+│     url = f"https://api.github.com/repos/{repository}/issues"                                    │
+│     since = datetime.datetime.now() - datetime.timedelta(days=10)                                │
+│     params = {'since': since.isoformat()}                                                        │
+│     response = requests.get(url, headers=headers, params=params)                                 │
+│     issues = response.json()                                                                     │
+│     return [(i['title'], i['number'], i['created_at']) for i in issues if i['number'] % 2 != 0]  │
+│                                                                                                  │
+╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
+What code feedback would you like to provide? (leave empty for approval):
+```
+
+Already after some quick initial feedback we can see that the LLM
+takes into account the users needs. After user provided feedback the
+automated tests are run. If those tests fail or run into exceptions
+they will be automatically passed back to the LLM for corrections.
+
+```shell
```

Comment:
Does this show the tests passing?  It's not obvious to me that the code shows pass or fail.